apps/apps/interactive-download-settings.js

```diff
@@ -1,21 +1,31 @@
-const msRestNodeAuth = require("@azure/ms-rest-nodeauth");
-const { WebSiteManagementClient, We } = require("@azure/arm-appservice");
+const { InteractiveBrowserCredential } = require("@azure/identity");
+const { WebSiteManagementClient } = require("@azure/arm-appservice");
 
 // CHANGE THESE VALUES TO YOUR OWN
-// Either set environment variables 
+// Either set environment variables
 // or set your value to the string at the end
 const subscriptionId = process.env["MY-SUBSCRIPTION"] || "";
 const resourceGroupName = process.env["MY-RESOURCE-GROUP"] || "";
 const resourceName = process.env["MY-RESOURCE-NAME"] || "";
 
-const getSettings = async (creds) => {
-    const client = new WebSiteManagementClient(creds, subscriptionId);
-    const settingsList = await client.webApps.listApplicationSettings(resourceGroupName, resourceName);
-    console.log(JSON.stringify(settingsList));
+const creds = new InteractiveBrowserCredential();

```

Comment:
By putting interactive above, it enters getSettings, even if the creds failed. Why do that - it is confusing and unclear to a user new to Azure when to examine the validity of the credentials.

Reply:
this is because @azure/identity doesn't return a promise for new InteractiveBrowserCredential() or new DefaultAzureCredential(), if we keep the original way, we can write a function 
```
function loginWithBrowser() { // or loginWithDefault for DefaultAzureCredential 
  try {
    return new InteractiveBrowserCredential();
  }
  catch(err) {
    throw err;
  }
}
```
and then call it like the orignal way 

```
loginWithBrowser().then(creds => {
  return getSettings(creds);
}) .catch(err => {
  console.log(err);
});
```
Let me know if you are okay with the proposed change ?

Reply:
@qiaozha 

 It sounds like your change is the right approach -please keep it.